esmvaltool/preprocessor/_multimodel.py

```diff
@@ -13,42 +26,330 @@
+    if t_axis is not None:
+        times = iris.coords.DimCoord(
+            t_axis, standard_name='time',
+            units=template_cube.coord('time').units)
+    lats = template_cube.coord('latitude')
+    lons = template_cube.coord('longitude')
+
+    # no plevs
+    if len(template_cube.shape) == 3:
+        cspec = [(times, 0), (lats, 1), (lons, 2)]
+    # plevs
+    elif len(template_cube.shape) == 4:
+        plev = template_cube.coord('air_pressure')
+        cspec = [(times, 0), (plev, 1), (lats, 2), (lons, 3)]
+
+    # correct dspec if necessary
+    fixed_dspec = np.ma.fix_invalid(cube_data,
+                                    copy=False,
+                                    fill_value=1e+20)
+    # put in cube
+    stats_cube = iris.cube.Cube(fixed_dspec,
+                                dim_coords_and_dims=cspec,
+                                long_name=stat_name)
+    coord_names = [coord.name() for coord in template_cube.coords()]
+    if 'air_pressure' in coord_names:
+        if len(template_cube.shape) == 3:
+            stats_cube.add_aux_coord(template_cube.
+                                     coord('air_pressure'))
+    stats_cube.attributes['_filename'] = file_name
+    # complete metadata
+    stats_cube.var_name = template_cube.var_name
+    stats_cube.long_name = template_cube.long_name
+    stats_cube.standard_name = template_cube.standard_name
+    stats_cube.units = template_cube.units
     return stats_cube
 
 
-def _sdat(srl_no):
-    """convert to a datatime point"""
-    new_date = dd(1950, 1, 1, 0) + td(srl_no)
+def _to_datetime(delta_t, unit_type):
+    """Convert to a datetime point"""
+    leap_correction = delta_t / (365. * 4.)
+    delta_t = delta_t + leap_correction
+    if unit_type == 'day since 1950-01-01 00:00:00.0000000':
+        new_date = datetime(1950, 1, 1, 0) + timedelta(np.int(delta_t))
+    elif unit_type == 'day since 1850-01-01 00:00:00.0000000':
+        new_date = datetime(1850, 1, 1, 0) + timedelta(np.int(delta_t))
+    # add more supported units here
     return new_date
 
 
 def _get_overlap(cubes):
-    """Get discrete time overlaps."""
-    all_times = [cube.coord('time').points for cube in cubes]
+    """
+    Get discrete time overlaps.
+
+    This method gets the bounds of coord time
+    from the cube and assembles a continuous time
+    axis with smallest unit 1; then it finds the
+    overlaps by doing a 1-dim intersect;
+    takes the floor of first date and
+    ceil of last date.
+    """
+    all_times = []
+    for cube in cubes:
+        # monthly data ONLY
+        # converts all to 365-days calendars
+        time_units = cube.coord('time').units
+        bnd1 = float(cube.coord('time').points[0])
+        bnd2 = float(cube.coord('time').points[-1])
+        if time_units.calendar is not None:
+            if time_units.calendar == '360_day':
+                bnd1 = int(bnd1 / 360.) * 365.
+                bnd2 = (int(bnd2 / 360.) + 1) * 365.
+            else:
```

Comment:
The code on line 191 and 192 is identical to the code on 195 and 196. Try to avoid code duplication.

Reply:
@bouweandela these calendar conversions will all disappear when I implement the usage of netcdf4 num2date functionality; left it like this for now, but this will change soon

Reply:
See #268